src/main/java/zbot/Parser.java

```diff
@@ -0,0 +1,144 @@
+package zbot;
+
+import zbot.exceptions.IncorrectInputException;
+import zbot.exceptions.InvalidCommandException;
+import zbot.exceptions.InvalidTaskNumberException;
+import zbot.exceptions.EmptyTaskListException;
+
+import zbot.tasks.TaskList;
+
+/**
+ * Represents a class that handles input parsing and executes the corresponding logic.
+ * <p>
+ * This class is responsible for:
+ * <ul>
+ *   <li>Parsing user input commands or data</li>
+ *   <li>Executing the appropriate actions based on the parsed input</li>
+ *   <li>Managing the flow of operations in response to the input</li>
+ * </ul>
+ * The class may interface with other components of the application to perform tasks, such as
+ * task management, file handling, or displaying results.
+ */
+class Parser {
+    /**
+     * Parses the input and executes the corresponding logic.
+     * <p>
+     * This method processes the user's input and triggers the appropriate actions based on the command.
+     * It interacts with the UI and task list to carry out the task.
+     * </p>
+     *
+     * @param input the input string entered by the user, containing the command and related arguments
+     * @param ui the UI object used to display messages and interact with the user
+     * @param taskList the list of tasks to be managed and modified based on the input
+     * @throws IncorrectInputException if the input does not follow the correct format
+     * @throws InvalidCommandException if the command entered by the user does not exist or is not recognized
+     * @throws EmptyTaskListException if the task list is empty and a task operation is attempted
+     * @throws InvalidTaskNumberException if the task number is less than 0 or greater than or equal to the size of the task list
+     */
+    public static void parseInput(String input, Ui ui, TaskList taskList) throws IncorrectInputException, InvalidCommandException, EmptyTaskListException, InvalidTaskNumberException {
+        String[] parts = input.split(" ", 2);
+        final String SUPPORTED_COMMANDS = "- list\n- mark\n- unmark\n- find\n- delete\n- todo\n- deadline\n- event\n- bye";
+        switch (parts[0]) {
+        case "list":
+            if (parts.length == 1) {
+                ui.showContents(taskList);
+            } else {
+                throw new IncorrectInputException("Sorry!! Please ensure your command matches the following example and have nothing else. " +
+                        "(e.g. \"list\")");
+            }
+            break;
+        case "delete":
+            if (parts.length != 2) {
+                throw new IncorrectInputException("Sorry!! Please ensure your command matches the example: \"delete 1\"");
+            }
+            int deleteIndex = getIndex(parts,taskList);
```

Comment:
coding standard: there should be a space between the parameters

```suggestion
            int deleteIndex = getIndex(parts, taskList);
```

Reply:
Thanks for the input! fixed this:)